vividus-extension-selenium/src/main/java/org/vividus/selenium/DriverFactory.java

```diff
@@ -0,0 +1,166 @@
+        () -> getCapabilitiesByPrefix(SELENIUM_GRID_PROPERTY_PREFIX));
+
+    private final Supplier<DesiredCapabilities> localDriverDesiredCapabilities = Suppliers.memoize(
+        () -> getCapabilitiesByPrefix(LOCAL_DRIVER_PROPERTY_PREFIX));
+
+    private final LoadingCache<Boolean, DesiredCapabilities> webDriverCapabilities = CacheBuilder.newBuilder()
+            .build(new CacheLoader<Boolean, DesiredCapabilities>()
+            {
+                @Override
+                public DesiredCapabilities load(Boolean local)
+                {
+                    DesiredCapabilities localCapabilities = localDriverDesiredCapabilities.get();
+                    if (Boolean.TRUE.equals(local))
+                    {
+                        return localCapabilities;
+                    }
+                    return merge(localCapabilities, seleniumGridDesiredCapabilities.get());
+                }
+            });
+
+    public DriverFactory(IRemoteWebDriverFactory remoteWebDriverFactory, IPropertyParser propertyParser,
+            JsonUtils jsonUtils)
+    {
+        this.remoteWebDriverFactory = remoteWebDriverFactory;
+        this.propertyParser = propertyParser;
+        this.jsonUtils = jsonUtils;
+    }
+
+    private DesiredCapabilities getCapabilitiesByPrefix(String prefix)
+    {
+        return propertyParser.getPropertyValuesTreeByPrefix(prefix)
+            .entrySet()
+            .stream()
+            .map(e -> isBoolean(e.getValue())
+                ? Map.entry(e.getKey(), Boolean.parseBoolean((String) e.getValue()))
+                : e)
+            .collect(Collectors.collectingAndThen(toMap(Entry::getKey, Entry::getValue), DesiredCapabilities::new));
+    }
+
+    private boolean isBoolean(Object value)
+    {
+        return value instanceof String && equalsAnyIgnoreCase((String) value, "true", "false");
+    }
+
+    @Override
+    public WebDriver getRemoteWebDriver(DesiredCapabilities desiredCapabilities)
+    {
+        DesiredCapabilities mergedDesiredCapabilities = getWebDriverCapabilities(false, desiredCapabilities);
+        return createWebDriver(remoteWebDriverFactory.getRemoteWebDriver(remoteDriverUrl,
+                updateDesiredCapabilities(mergedDesiredCapabilities)));
+    }
+
+    protected WebDriver createWebDriver(WebDriver webDriver)
+    {
+        WebDriver driver = new TextFormattingWebDriver(webDriver);
+        configureWebDriver(driver);
+
+        LOGGER.atInfo()
+              .addArgument(() -> jsonUtils
+                      .toPrettyJson(WebDriverUtil.unwrap(driver, HasCapabilities.class).getCapabilities().asMap()))
+              .log("Session capabilities:\n{}");
+        return driver;
+    }
+
+    protected DesiredCapabilities updateDesiredCapabilities(DesiredCapabilities desiredCapabilities)
+    {
+        return desiredCapabilities;
+    }
+
+    protected void configureWebDriver(WebDriver webDriver)
```

Comment:
I recommend to make this method abstract and keep empty implementation in child classes